int32 x
int32 y
int32 road_x
int32 road_y
